src/_guides/language/effective-dart/usage.md

```diff
@@ -938,16 +938,16 @@ set x(num value) => center = Point(value, center.y);
 {% endprettify %}
 
 
-### DON'T use `this.` when not needed to avoid shadowing.
+### DON'T use `this.` except to avoid shadowing or redirect to a named constructor.
 
 {% include linter-rule.html rule="unnecessary_this" %}
 
 JavaScript requires an explicit `this.` to refer to members on the object whose
 method is currently being executed, but Dart&mdash;like C++, Java, and
 C#&mdash;doesn't have that limitation.
 
-The only time you need to use `this.` is when a local variable with the same
-name shadows the member you want to access.
+There are only two times you need to use `this.`. One is when a local variable
```

Comment:
We generally avoid empty words like "There are". But I'm blanking on how to fix it, so ... let it be.